src/stats/stocks.ts

```diff
@@ -1,26 +1,42 @@
 // @flow
 import _ from 'lodash';
-import { SlippiGame } from "../SlippiGame";
-import { iterateFramesInOrder, isDead, didLoseStock } from "./common";
 
-import { StockType } from "./common";
+import { FrameEntryType, FramesType, isDead, didLoseStock, PlayerIndexedType, StockType } from "./common";
+import { StatComputer } from './stats';
 
-export function generateStocks(game: SlippiGame): StockType[] {
-  const stocks: Array<StockType> = [];
-  const frames = game.getFrames();
+interface StockState {
+  stock: StockType | null | undefined;
+}
+
+export class StockComputer implements StatComputer<StockType[]> {
+  private state = new Map<PlayerIndexedType, StockState>();
+  private opponentIndices = new Array<PlayerIndexedType>();
+  private stocks = new Array<StockType>();
+
+  public setPlayerIndices(playerIndices: PlayerIndexedType[]): void {
+    this.opponentIndices = playerIndices;
```

Comment:
The naming convention for these really confuses me in a lot of places. `playerIndices` gets set to `opponentIndices`? Truth is I struggled with naming these too. Really when it's in the array form its like `playerPermutations` or something, then when it's the single object it's `indices` maybe?

Reply:
Yeah you're telling me. I'm just as confused. The function `getSinglesOpponentIndices()` returns a type called `PlayerIndexedType`. I don't know whether it should be called player indices or opponent indices. In any case I've just named them all `opponentIndices` for now. Fixed in commit 85d076.

Reply:
No particular preference. Preferable if names are something short and succinct.